### PR TITLE
[new release] reason and rtop (3.8.0)

### PR DESCRIPTION
**Pull request by anmonteiro**
Reason: Syntax & Toolchain for OCaml

- Project page: <a href="https://github.com/reasonml/reason">https://github.com/reasonml/reason</a>
- Documentation: <a href="https://reasonml.github.io/">https://reasonml.github.io/</a>

##### CHANGES:

- Add support for OCaml 4.13 (@EduardoRFS and @anmonteiro) [#2657](https://github.com/reasonml/reason/pull/2657)
- Add support for OCaml 4.14 (@EduardoRFS and @anmonteiro) [#2662](https://github.com/reasonml/reason/pull/2662)

### CODE REVIEW

**Comment by anmonteiro** [comment]
I fixed the dependency bound issues in https://github.com/reasonml/reason/pull/2664.

File: packages/reason/reason.3.8.0/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "ocaml"         {>= "4.03" & < "4.15"}
+  "dune"          {>= "1.4"}
+  "ocamlfind"     {build}
+  "menhir"        {>= "20180523"}
+  "merlin-extend" {>= "0.6"}
+  "fix"
+  "result"
+  "ppx_derivers"
+]
+synopsis: "Reason: Syntax & Toolchain for OCaml"
```
Comment (mseri) [inline packages/reason/reason.3.8.0/opam:23]:
```suggestion
conflicts: [ "base-effects" ]
synopsis: "Reason: Syntax & Toolchain for OCaml"
```

Comment (mseri) [reply packages/reason/reason.3.8.0/opam:23]: Due to 
```
#=== ERROR while compiling reason.3.8.0 =======================================#
# context              2.1.2 | linux/x86_64 | ocaml-variants.4.12.0+domains+effects | pinned(https://github.com/reasonml/reason/releases/download/3.8.0/reason-3.8.0.tbz)
# path                 ~/.opam/4.12/.opam-switch/build/reason.3.8.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p reason -j 31
# exit-code            1
# env-file             ~/.opam/log/reason-15391-4ebce0.env
# output-file          ~/.opam/log/reason-15391-4ebce0.out
### output ###
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -w -9 -g -bin-annot -I src/vendored-ocaml-migrate-parsetree-v1.8.0/src/.reason_migrate_parsetree.objs/byte -I /home/opam/.opam/4.12/lib/ocaml/compiler-libs -I /home/opam/.opam/4.12/lib/ppx_derivers -I /home/opam/.opam/4.12/lib/result -no-alias-deps -open Reason_migrate_parsetree__ -o src/vendored-ocaml-migrate-parsetree-v1.8.0/src/.reason_migrate_parsetree.objs/byte/reason_migrate_parsetree__Ast_412.cmo -c -impl src/vendored-ocaml-migrate-parsetree-v1.8.0/src/ast_412.pp.ml)
# File "src/vendored-ocaml-migrate-parsetree-v1.8.0/src/ast_412.ml", lines 246-301, characters 2-44:
# Error: This variant or record definition does not match that of type
#          Parsetree.pattern_desc
#        Constructors number 17 have different names, Ppat_effect and Ppat_extension.
```

**Comment by mseri** [comment]
Thanks

**Pull request merged**